### ISSUE TITLE
[1.19.2] Cancelling `EntityInteractSpecific` event does not cancel interaction server-side

### ISSUE THREAD

**Issue by Fuzss** [MinecraftForge/MinecraftForge#9040]
**Minecraft Version:** 1.19.2

**Forge Version:** 43.1.25

**Steps to Reproduce:**
 1. Cancel `PlayerInteractEvent.EntityInteractSpecific` by providing a cancellation result and setting `canceled` to `true`
 2. Vanilla processing for the event will still take place, and code provided in the event will be run afterwards

**Example:**
Example for cancelling interactions with armor stand, theoretically this should prevent any armor from being added or taken. In reality nothing is changed.
```java
    @SubscribeEvent
    public static void onEntityInteractSpecific(final PlayerInteractEvent.EntityInteractSpecific evt) {
        if (evt.getTarget() instanceof ArmorStand) {
            evt.setCancellationResult(InteractionResult.sidedSuccess(evt.getLevel().isClientSide));
            evt.setCanceled(true);
        }
    }
```
**Description of issue:**
The problem is the Forge event is fired too late in `ServerGamePacketListenerImpl`, only after the vanilla interaction has been processed. Simply moving the event invocation to before vanilla's code is not enough though, currently the event completely ignores the provided cancellation result.

Therefore I suggest when the event is cancelled setting the result to the local `InteractionResult` variable, otherwise let vanilla run and set the variable itself. The local variable is later used by vanilla for triggering a statistics tracker as well as calling the arm swing server side for other players to see.

**Comment by Fuzss** [comment]
Looking a bit more at this, there is another issue: `EntityInteractSpecific::getLocalPos` is supposed to provide the hit vector (according to the events javadoc), aka where on the bounding box of the entity the player has clicked. Instead `Entity::position` is provided as the local position, which is well, wrong.

It should also be noted that all this, also the main issue described above only affects the server, everything is implemented properly client-side.

To fix all those issues I'd suggest moving the event invocation to the head of the lambda called in `ServerboundInteractPacket.Handler::onInteraction`. The hit vector is only available in this method, and not where the current Forge hook is. Fabric Api does it similarly, and that implementation works.

**Comment by Lupicus** [comment]
#8143

**Comment by MagnusHJensen** [comment]
> Looking a bit more at this, there is another issue: `EntityInteractSpecific::getLocalPos` is supposed to provide the hit vector (according to the events javadoc), aka where on the bounding box of the entity the player has clicked. Instead `Entity::position` is provided as the local position, which is well, wrong.

Moving the Event call definitely worked, and passing the Vec3 inside `onInteraction` did indeed pass the proper local position.
However, I'm unsure how to go about doing this to avoid a breaking change, or does that not matter since it's a bugged value?
